src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.RenameTrackingCommitter.cs

```diff
@@ -27,30 +29,34 @@ private class RenameTrackingCommitter : ForegroundThreadAffinitizedObject
             private readonly IEnumerable<IRefactorNotifyService> _refactorNotifyServices;
             private readonly ITextUndoHistoryRegistry _undoHistoryRegistry;
             private readonly string _displayText;
-            private readonly bool _showPreview;
+
+            // This task performs the actual rename on solution. We kick this off during preview to be able to show the diff preview
+            // and if it was successfully completed without being cancelled, we reuse its result during the actual commit operation as well.
+            private Task<RenameTrackingSolutionSet> _renameSymbolTask;
```

Comment:
:sparkles:  :clap: Thanks, I was able to refactor and use `AsyncLazy`. It greatly reduced the amount of extra work I was doing to share the rename task result across preview/commit. `AsyncLazy` is :godmode: